### DISCUSSION TITLE
Connecting Design Systems Hack [Group 2] Natural language

### DISCUSSION THREAD

**Discussion by patrickpatrickpatrick** [alphagov/govuk-design-system#4333] [Community events]
Discussion for Group 2 of the Design System Hack Day - Linking Design Systems

**Comment by murrlipp** [comment]
Reasons for choosing:

- Tied into discoverability.  Would like to focus on the user needs - i.e. 'I want to alert a user to something' rather than 'I want to use a notification'.
- Making the naming of components and patterns more understandable - i.e. some people say dropdown, some people say 'select'.
- Confusion between the name of the component and a specific HTML attribute.
- To help solve the problem for people who are not designers, to break the jargon. They don't know what to search for.
- What would you search for if you didn't know how a certain design system has named their component?
- To help with people in international contexts.  In such a huge context, if you don't know what you don't know, how do you search for it?
- Also benefits new joiners in teams in instances where there are specific terms used.

**Comment by murrlipp** [comment]
How might we...
- Communicate to people who want to solve a user problem with a design system component or pattern, what user need the component or pattern solves.
- Make this discoverable across all departmental design systems.

**Comment by murrlipp** [comment]
Thoughts from the IRL group:

--

GOV.UK and most other design systems DO have a sentence on component pages that accurately summarises what user need is met by the component, but this isn't at the forefront so people still need to scroll through all of the components to find the one they need.

We want to make the process of searching for a relevant component pattern easier by introducing an LLM-fed search pattern on the GOV.UK Design System website (https://design-system.service.gov.uk/)

This would allow users to enter a natural language search query such as 'I want to alert a user to an important piece of information', and they would see a search results page showing relevant components for the user need they have entered.

An extension to this could be that the search results could also include other design systems - for discoverability.

--

ChatGPT does this, so could it be integrated into GOV.UK?

![Screenshot 2024-11-28 at 14 38 16](https://github.com/user-attachments/assets/cbfcf74b-2de7-4964-bf2c-049f14e6ac12)

**Comment by murrlipp** [comment]
A mockup of what a search results page might look like:

![image](https://github.com/user-attachments/assets/1668b7ae-d3a1-4da6-80eb-c63ed32c3a36)

**Comment by murrlipp** [comment]
# Summary

## Exploring: Use of natural language to describe components

### Why?
- Tied into discoverability. Would like to focus on the user needs - i.e. 'I want to alert a user to something' rather than 'I want to use a notification'.
- Making the naming of components and patterns more understandable - i.e. some people say dropdown, some people say 'select'.
- To help solve the problem for people who are not designers, to break the jargon. They don't know what to search for.
- What would you search for if you didn't know how a certain design system has named their component?
- To help with people in international contexts. In such a huge context, if you don't know what you don't know, how do you search for it?
- Also benefits new joiners in teams in instances where there are specific terms used.

### Concept

GOV.UK and most other design systems have a sentence on component pages that accurately summarises what user need is met by the component, but this isn't at the forefront so people still need to scroll through all of the components to find the one they need.

GOV.UK recently updated the search bar on their main site to allow more searches in more natural language.

We want to make the process of searching for a relevant component pattern easier by introducing an LLM-fed search pattern on the GOV.UK Design System website (https://design-system.service.gov.uk/)

This would allow users to enter a natural language search query such as 'I want to alert a user to an important piece of information', and they would see a search results page showing relevant components for the user need they have entered.

An extension to this could be that the search results could also include other design systems - for discoverability.

### Mockup

![image](https://github.com/user-attachments/assets/13f422ca-21a0-4ab7-b6d3-b18638e3340d)

**Comment by ImranH-GDS** [comment]
Hello, we have about 30 minutes left. Can each group/ subgroup think about who will play back your ideas during the show and tell?

**Comment by designwithcal** [comment]
Hello from the other side. We've been busy sketching out a few wireframes of a solution. It's all on Figma here: https://www.figma.com/board/9HMSap9Ec50uqGzfshDQcv/Group-2---Natural-Language?node-id=0-1&t=oPnECoslcBZUDtlO-1

**Comment by onupama1** [comment]
Hello @ImranH-GDS my contributions re bonkers ideas for language:
Ferdinand de Sassure says language is a system of signs. Signifier = the word, image or symbol we see or hear. e.g. word apple
signified = the idea or concept it represents. e.g. the fruit apple we eat
for neurodiversity the signifier does not always make the concept or signified otherwise obvious for others. We don't remember always or think differently or recognise things in patterns or visually etc not just text.
I do not have the exact solution now re how to help discoverability in DS. but maybe think beyond menus. 
e.g. 1: grouped sections or colour coded categories. Users can recognise what they need without remembering words. 
e.g. 2: the search system can suggest the icons and words both together like money icon can be associated with benefits or grants or loans
e.g.3: multiple entry points that is not just the menu but icons or colour category
e.g. 4: reminder or pinning pages. Help users remember where they have been
e.g. 5: help users have a chatbot or bot where they don't remember the words but can describe what they are looking for. help them concretise their thoughts

**Comment by HithertoAI** [comment]
[Connecting Design Systems 'Hack' #2.pdf](https://github.com/user-attachments/files/25210905/Connecting.Design.Systems.Hack.2.pdf)


We weren't able to present our idea today. It's rather comprehensive and has potential for expansion. I've uploaded the PDF of our board. 

https://miro.com/app/board/uXjVGGK8VN0=/?moveToWidget=3458764658912888467&cot=14

**Comment by NaomiGrantTPX** [comment]
Hi @ImranH-GDS, @emma-nicol 

Breakout room 4 - _**How might we find what they need through language they naturally use?**_

Our real world example was initially informed by EasyJet’s 'Inspire me' tool 🧡

1. **A series of function-specific case studies** that centre around storytelling that uses general language. Users know what new service they want to develop but they may not be aware of all of the assets that have already been developed. The user knows what the main function is **_(example: a member of the public applying for a new Blue badge)_**. Case studies would outline 1. what tasks the end user might need to step through, 2. behaviours they might exhibit, 3. what information they might need to provide etc. Then would signpost to the various existing design systems and include component and pattern suggestions based on the best practice and the criteria above.

2. Continuous improvement idea: **A decision tree tool**. Users would answer binary questions that use general language to create a specific story. A results page would map out a high-level end user journey which would signpost out to the various existing design systems and includes component and pattern suggestions based on the criteria above and best practice
